codalab/server/bundle_manager.py

```diff
@@ -281,17 +286,42 @@ def _make_bundle(self, bundle):
                             with open(dependency_path, 'wb') as f:
                                 shutil.copyfileobj(fileobj, f)
 
+                    # If source is local file system and destination is blob storage:
+                    # need to copy everything into a temp folder and upload together
+                    elif parse_linked_bundle_url(path).uses_beam:
+                        if dep.child_path != "":
+                            tempdir_dependency_path = os.path.join(tempdir, dep.child_path)
+                        else:
+                            tempdir_dependency_path = os.path.join(tempdir, dep.parent_uuid)
+                        path_util.copy(
+                            dependency_path, tempdir_dependency_path, follow_symlinks=False
+                        )
+                        dependency_path = tempdir_dependency_path
                     deps.append((dependency_path, child_path))
+                remove_path(path)  # delete the original bundle path
+
+                # Upload to destination bundle storage
+                if parse_linked_bundle_url(path).uses_beam:  # The destination is using blob storage
+                    source_fileobj = zip_util.tar_gzip_directory(
+                        tempdir
+                    )  # pack all the files in temp folder to a fileobj
+                    source_filename = "MakeBundle.tar.gz"  # TODO(Jiani): Use the original filename
```

Comment:
should we fix this for this PR? and does this file name even matter / does it get saved in the output bundle?

Reply:
This filename ("MakeBundle.tar.gz") will not be save in the output bundle.  However, the output bundle will automatically be named as `make-[source bundle id]`. I will remove this comment. Thanks Ashwin!

Reply:
@wwwjn can you remove the comment, and also add another comment noting the filename doesn't make a difference / isn't saved in the output?